src/main/java/org/spongepowered/api/entity/ai/Goal.java

```diff
@@ -64,15 +64,15 @@
      * @param task The task to run
      * @return This goal, for chaining
      */
-    Goal<O> addTask(int priority, AITask<? extends O> task);
+    Goal<O> addTask(int priority, AITask<? super O> task);
```

Comment:
A task appliable to a `Zombie` may not be applicable to `ZombiePigman`. In `AITask` interface definition, the `<O>` already extends `Agent`.